### ISSUE TITLE
Create clap command `ockam tcp-outlet delete` to delete a TCP Outlet on a node

### ISSUE THREAD

**Issue by caoakleyii** [build-trust/ockam#4268]
## Summary

A node has set of TCP Outlets. This command should delete an outlet of a specific alias
```
ockam tcp-outlet delete [alias] --node [node-name]
```
The command should have a positional argument `[alias]` which is the alias of the tcp outlet to be deleted.  
The command should accept a `--node` flag, however default to the default node if one is not provided.

Code of where outlets are created and stored can be found [here](https://github.com/build-trust/ockam/blob/65c7259a2b96e85e2745397eaaf77e7dcf427960/implementations/rust/ockam/ockam_api/src/nodes/service/portals.rs#L257-L323).

`tcp-outlet` commands are defined [here](https://github.com/build-trust/ockam/blob/develop/implementations/rust/ockam/ockam_command/src/tcp/outlet/mod.rs).
## Expected Behavior

```sh
ockam tcp-outlet create --at /node/n1 --from /service/outlet --to 127.0.0.1:5000 --alias "my-outlet"
...
ockam tcp-outlet delete my-outlet --node n1
Deleted TCP Outlet 'my-outlet' on node 'n1'
```

_We love helping new contributors!_
_If you have questions or need help as you work on your first Ockam contribution, please leave a comment on [this discussion](https://github.com/ockam-network/ockam/discussions/1081)._
_If you're looking for other issues to contribute to, checkout [this discussion](https://github.com/build-trust/ockam/discussions/3206) and labels -  https://github.com/build-trust/ockam/labels/good%20first%20issue or https://github.com/build-trust/ockam/labels/help%20wanted_

**Comment by nidnogg** [comment]
Hiya @CAOakleyII, I'll take a stab at this and the sibling issue https://github.com/build-trust/ockam/issues/4269. If you could assign me to those whenever possible, that'd be great!

**Comment by nidnogg** [comment]
Hey, I've had a chance to look over these two issues for some time, but I've gotten somewhat stumped over the last few steps. I've added a draft PR https://github.com/build-trust/ockam/pull/4341 (for the TCP outlet side) to hopefully get some guidance. The TCP inlet PR should come shortly after this one is solved.

A couple of questions regarding that:
- Whenever I run `ockam tcp-outlet delete my-outlet --node n1`, I get an error like the following:
```
An error occurred while processing the request. Status code: 400 BadRequest. Message: Invalid endpoint: /node/outlet
```
I've put up new `Delete` endpoint at `/node/outlet` like [this](https://github.com/build-trust/ockam/pull/4341/commits/7ee3117e9c03d26860125b65322c828465948bbf#diff-fee061b9c2629c91dcb8dcd8412526980626c18ba53562340f2040cab470469aL599-L600). However, I'm not sure calling `Request::delete('/node/outlet')` like I do [here](https://github.com/build-trust/ockam/pull/4341/commits/7ee3117e9c03d26860125b65322c828465948bbf#diff-1c7ebe10ced1a6d919c6ccf4dea6056d7b204651698520b69eb41e7b46f34d96R81-R83) is enough to reach it.

- Is calling `stop_outlet()` and removing the alias key from the `BTree` node registry enough for deletion? [This](https://github.com/build-trust/ockam/pull/4341/commits/7ee3117e9c03d26860125b65322c828465948bbf#diff-5b3f34a98f0adf81c262908c1b6ee444f6e61e4242075b18860c9a57ea986e3fR338-R351) is how I'm currently planning on doing this.

**Comment by davide-baldo** [comment]
Hi @nidnogg! :wave: 
I've added a few comments to your draft PR! they should be enough to guide you through the next steps.

Regarding this error
>  ```An error occurred while processing the request. Status code: 400 BadRequest. Message: Invalid endpoint: /node/outlet```

It's most likely due to another version of ockam running in the background, always make sure you are running your version: first stop/delete currently running ockam processes via `ockam node stop` and `ockam node delete` or use `ockam reset` if you feel lazy :smile:  and then create/start your node again!

**Comment by nidnogg** [comment]
> Hi @nidnogg! 👋 I've added a few comments to your draft PR! they should be enough to guide you through the next steps.
> 
> Regarding this error
> 
> > `An error occurred while processing the request. Status code: 400 BadRequest. Message: Invalid endpoint: /node/outlet`
> 
> It's most likely due to another version of ockam running in the background, always make sure you are running your version: first stop/delete currently running ockam processes via `ockam node stop` and `ockam node delete` or use `ockam reset` if you feel lazy 😄 and then create/start your node again!

Hey @davide-baldo, thanks for hopping by and helping out! I've incorporated a couple of the suggestions (specifically the type tag which I'd forgotten about and the actual printed output) but I'm still unable to reach the API for some reason. 

I had a previous ockam installation from brew which I did remove, as well as the `~/.ockam` folder. I did `cargo run --bin ockam reset` and it's still hitting that 400 BadRequest error. 

I also left some replies to your comments regarding `worker_addr`.

**Comment by davide-baldo** [comment]
>  hitting that 400 BadRequest error

Are you hitting a generic `BadRequest` or specifically ```Status code: 400 BadRequest. Message: Invalid endpoint: /node/outlet```?
The latter would be very strange (I've tested your code, it works!!), you can try to run the node in foreground with trace logging enabled:

```ockam node create --foreground -vvv my-node-name```

and see what the log says about the error.

**Comment by nidnogg** [comment]
> Are you hitting a generic BadRequest or specifically Status code: 400 BadRequest. Message: Invalid endpoint: /node/outlet?

Interestingly enough, I'm sure it used to be a Invalid endpoint one - but once I experimented around with the code a bit it turned into the generic `BadRequest` error. 

Thanks to your `node create --foreground -vvv` suggestion though I've managed to finally delete the outlets properly, although I feel like displaying an error in case a non-existent `alias` is provided would be helpful to the end-user. Right now it just displays that `BadRequest` error when this situation happens.

As it has been a while on this issue though, I'll push the changes without the error logging on the CLI side in case they are better reserved for a future issue. 

Also started working on the sibling TCP inlet code, another PR should be available soon.

EDIT: Added some more changes requested by @adrianbenavides, handling errors a bit better now based on the response status codes
